sagemaker-builder/packing-a-tensorflow-model/packing_your_tensorflow_model.ipynb

```diff
@@ -0,0 +1,783 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Packing your TensorFlow model\n",
+    "\n",
+    "With Amazon SageMaker, you can package your own algorithms that can then be trained and deployed in the SageMaker environment. This notebook guides you through an example using TensorFlow that shows you how to build a container for SageMaker and use it for training.\n",
+    "\n",
+    "## Permissions\n",
+    "\n",
+    "Running this notebook requires permissions in addition to the normal `SageMakerFullAccess` permissions. This is because it creates new repositories in Amazon ECR. The easiest way to add these permissions is simply to add the managed policy `AmazonEC2ContainerRegistryFullAccess` to the role that you used to start your notebook instance. There's no need to restart your notebook instance when you do this, the new permissions will be available immediately.\n",
+    "\n",
+    "## The example\n",
+    "\n",
+    "In this example we show how to package a custom TensorFlow algorithm with a Python example which works with the CIFAR-10 dataset: [CIFAR-10]: http://www.cs.toronto.edu/~kriz/cifar.html"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The parts of the algo\n",
+    "\n",
+    "The `src` directory has all the components you need to package the sample algorithm for Amazon SageMager:\n",
+    "\n",
+    "    └── src\n",
+    "        ├── cifar10.py\n",
+    "        └── resnet_model.py\n",
+    "\n",
+    "Let's discuss each of these in turn:\n",
+    "\n",
+    "* __`src`__ is the directory which contains the files that are installed in the container.\n",
+    "\n",
+    "The files that we put in the container are:\n",
+    "\n",
+    "* __`cifar10.py`__ is the program that implements our training algorithm.\n",
+    "* __`resnet_model.py`__ is the program that contains our Resnet model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Packing the Training Code"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "FROM tensorflow/tensorflow:1.11.0-py3\n",
+      "\n",
+      "RUN apt-get update && apt-get install -y --no-install-recommends git\n",
+      "\n",
+      "RUN git clone https://github.com/mvsusp/sagemaker-containers.git -b mvs-sagemaker-containers-train-improvements && cd sagemaker-containers && pip install . --quiet --disable-pip-version-check\n",
```

Comment:
are these changes that we can get into master so the example doesn't have to refer to branch that will get stale quickly?

Reply:
I wanted you to review the experience before submitting the PR with changes to sagemaker containers. I will not push that without these changes being done first.